### ISSUE TITLE
ArgumentOutOfRangeException  @ PSConsoleReadLine

### ISSUE THREAD

**Issue by vivekragunathan** [PowerShell/PSReadLine#1568]
```
Oops, something went wrong.  Please report this bug with the details below.
Report on GitHub: https://github.com/lzybkr/PSReadLine/issues/new
-----------------------------------------------------------------------
Last 28 Keys:
 UpArrow UpArrow UpArrow UpArrow UpArrow UpArrow DownArrow DownArrow DownArrow DownArrow Enter
 UpArrow Enter
 w s l Space - - v e r s i o n Enter
 Ctrl+l

Exception:
System.ArgumentOutOfRangeException: The value must be greater than or equal to zero and less than the console's buffer size in that dimension.
Parameter name: top
Actual value was -1.
   at System.Console.SetCursorPosition(Int32 left, Int32 top)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(ConsoleKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics)
-----------------------------------------------------------------------
```

**Comment by davinci26** [comment]
I run into the same issue here with the following error:

### Environment
PSReadLine: 2.0.0-rc1
PowerShell: 5.1.19041.1
OS: Microsoft Windows 10.0.19041
Last 200 Keys
```
 LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow Backspace Backspace Backspace Backspace Backspace Backspace N a Tab Enter
 G E t - Backspace Backspace Backspace e t 0 Backspace - V a r i a b l e Spacebar * D Backspace D e v i c e Enter
 UpArrow Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace D Backspace Spacebar D e v c e Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace b l e Spacebar * d Backspace - D e v Enter
 UpArrow LeftArrow LeftArrow RightArrow RightArrow LeftArrow RightArrow LeftArrow LeftArrow LeftArrow Backspace RightArrow RightArrow RightArrow i c e * Enter
 UpArrow UpArrow UpArrow UpArrow UpArrow DownArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow RightArrow RightArrow RightArrow RightArrow RightArrow RightArrow RightArrow RightArrow RightArrow RightArrow RightArrow RightArrow RightArrow RightArrow RightArrow Spacebar = Backspace - A d d Tab Spacebar " " LeftArrow t 
e Backspace Backspace T e Backspace Backspace / t e s t m o d e : l o o p Spacebar Backspace Enter
 UpArrow Enter
 UpArrow DownArrow UpArrow UpArrow
```

### Exception
```
System.ArgumentOutOfRangeException: The value must be greater than or equal to zero and less than the console's buffer size in that dimension.
Parameter name: top
Actual value was 12.
   at System.Console.SetCursorPosition(Int32 left, Int32 top)
   at Microsoft.PowerShell.Internal.VirtualTerminal.SetCursorPosition(Int32 left, Int32 top)
   at Microsoft.PowerShell.PSConsoleReadLine.ReallyRender(RenderData renderData, String defaultColor)
   at Microsoft.PowerShell.PSConsoleReadLine.ForceRender()
   at Microsoft.PowerShell.PSConsoleReadLine.Render()
   at Microsoft.PowerShell.PSConsoleReadLine.UpdateFromHistory(HistoryMoveCursor moveCursor)
   at Microsoft.PowerShell.PSConsoleReadLine.HistoryRecall(Int32 direction)
   at Microsoft.PowerShell.PSConsoleReadLine.PreviousHistory(Nullable`1 key, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(PSKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics, CancellationToken cancellationToken)
```

**Comment by daxian-dbw** [comment]
@VivekRagunathan This issue was fixed (see #1315). Please upgrade to the 2.0.1 version of PSReadLine from PowerShellGallery. See the [upgrading section](https://github.com/PowerShell/PSReadLine#upgrading) for instructions.

@davinci26 The stack trace you reported is different, and it's likely fixed by #1306. Plase also upgrad to the 2.0.1 version of PSReadLine and give it a go. If you still able to repro after the upgrade, please open a new issue with the environment data and repro steps (when it repros with `UpArrow`, then the history command retrieved by that `UpArrow` should be reported here ideally).
